tutorials/ardu-pi-serial-part-1/index.md

```diff
@@ -0,0 +1,330 @@
+- Open the serial connection to the Arduino
+- Read the Sense HAT environmental sensors
+- Request the analog sensor readings from Arduino over the serial connection
+- Publish the sensor readings as a JSON document once a second
+- Re-generate and sign a new JWT token after it expires
+
+1. SSH to the Raspberry Pi with:
+```
+$ ssh pi@<your_Pi_IP>
+```
+1. Connect the Arduino to the Pi with a USB cable. This will power on the Arduino, and create a USB serial port on the Pi, for serial communications.
+1. On the Pi, check that you now have a serial port connected to the Pi. Execute: 
+```
+$ file /dev/ttyACM0
+```
+The output should be similar to:
+```
+/dev/ttyACM0: character special (166/0)
+```
+1. With your virtualenv active, run the client app on the Pi, by executing:
+```
+(venv) $ python ardu-pi-serial.py  \
+    --project_id your_project_id \
+    --registry_id my-registry \
+    --device_id my-device \
+    --private_key_file rsa_private.pem \
+    --algorithm RS256 \
+    --cloud_region us-central1 \
+    --ca_certs roots.pem \
+    --message_type event \
+    --device_type pi
+```
+To see the command line argument options, execute:
+```
+(venv) $ python ardu-pi-serial.py --help
+```
+If the client app starts correctly, you should see output similar to this:
+```
+Creating JWT using RS256 from private key file rsa_private.pem
+Creating and flushing serial port. Rebooting Arduino..
+('gcp_on_connect', '0: No error.')
+Received message '' on topic '/devices/my-device/config' with Qos 1
+Sleeping 3s..
+Received from Arduino: S s:89 w:42
+Publishing message: {"temperature": 37.32, "timestamp": "2018-07-25T08:29:59+08:00", "clientid": "raspberrypi", "humidity": 37.98, "pressure": 1016.98, "windgen": 0.21, "solargen": 0.43}
+gcp_on_publish
+```
+
+1. On your local development machine, create a subscription for your IoT Core registry's telemetry topic, by executing:
+```
+$ gcloud pubsub subscriptions create my-device-events-sub --topic=my-device-events
+```
+1. Verify that your telemetry data is streaming to Cloud Pub/Sub, by executing: 
+```
+$ gcloud pubsub subscriptions pull my-device-events-sub --limit 100 --auto-ack
+```
+You should see message payloads similar to:
+```
+{"temperature": 36.62, "timestamp": "2018-07-25T08:42:10+08:00", "clientid": "raspberrypi", "humidity": 38.44, "pressure": 1017.05, "windgen": 0.1, "solargen": 1.2}   │ 151338809303544 │ deviceId=my-device
+```
+Now your Arduino and Pi are streaming telemetry data through Cloud IoT Core to Pub/Sub. You should see the sensor readings change if you expose the solar panel to more, or less light. And the wind generator (DC motor) will start generating power, as you turn the rotor.
+
+## Next
+In part 2 (coming soon) of this tutorial, you will use Cloud Dataflow with a Python streaming pipeline, to process this data stream, store it in Cloud BigQuery, explore the data with a Cloud Datalab Jupyter Notebook, and visualize it using Cloud Data Studio.
```

Comment:
Strike the comma after the word "pipeline".

Reply:
Done